app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt

```diff
@@ -105,6 +117,19 @@ internal class VaultAccountsViewModel @Inject constructor(
             }
             accountsRepository
                 .loadAddresses(vaultId)
+                .onEach { addresses->
+                    addresses.forEach{
+                        val addressOrder = chainsOrderRepository.find(it.chain.raw)
+                        if (addressOrder == null)
+                            chainsOrderRepository.insert(it.chain.raw)
+                    }
+                }
+                .zip(chainsOrderRepository.loadByOrders()) { addresses, chainOrders ->
+                    chainOrders.map {orderEntity->
+                         addresses.find { address -> address.chain.raw == orderEntity.value }!!
+                    }
+                }
+                .flowOn(IO)
```

Comment:
Dispatcher should be defined inside the repository

Reply:
Resolved